modules/ppcp-compat/src/SettingsMapHelper.php

```diff
@@ -5,66 +5,155 @@
  * @package WooCommerce\PayPalCommerce\Compat
  */
 
-declare(strict_types=1);
+declare( strict_types = 1 );
 
 namespace WooCommerce\PayPalCommerce\Compat;
 
+use RuntimeException;
+
 /**
- * A helper for mapping the new/old settings.
+ * A helper class to manage the transition between legacy and new settings.
+ *
+ * This utility provides mapping from old setting keys to new ones and retrieves
+ * their corresponding values from the appropriate models. The class uses lazy
+ * loading and caching to optimize performance during runtime.
  */
 class SettingsMapHelper {
 
 	/**
-	 * A list of mapped settings.
+	 * A list of settings maps containing mapping definitions.
 	 *
 	 * @var SettingsMap[]
 	 */
 	protected array $settings_map;
 
+	/**
+	 * Indexed map for faster lookups, initialized lazily.
+	 *
+	 * @var array|null Associative array where old keys map to metadata.
+	 */
+	protected ?array $key_to_model = null;
+
+	/**
+	 * Cache for results of `to_array()` calls on models.
+	 *
+	 * @var array Associative array where keys are model IDs.
+	 */
+	protected array $model_cache = array();
+
 	/**
 	 * Constructor.
 	 *
-	 * @param SettingsMap[] $settings_map A list of mapped settings.
+	 * @throws RuntimeException When an old key has multiple mappings.
+	 *
+	 * @param SettingsMap[] $settings_map A list of settings maps containing key definitions.
```

Comment:
```suggestion
	 * @param SettingsMap[] $settings_map A list of settings maps containing key definitions.
	 *
	 * @throws RuntimeException When an old key has multiple mappings.
```

I think the `throws` tag should go after the `param` and `return` tags

Reply:
PhpStorm arranged those headers, but that's just my personal configuration 🙈

In most of the code the @throws tag comes even later, after the @return, so I've updated my PhpStorm settings accordingly